src/realm/query_expression.hpp

```diff
@@ -3028,8 +3028,42 @@ class Columns<Lst<T>> : public ColumnsCollection<T> {
     {
         return make_subexpr<Columns<Lst<T>>>(*this);
     }
-    friend class Table;
-    friend class LinkChain;
+};
+
+template <>
+class Columns<Lst<String>> : public ColumnsCollection<String> {
+public:
+    using ColumnsCollection<String>::ColumnsCollection;
+    using ColumnListBase::m_column_key;
+    using ColumnListBase::m_link_map;
+
+    std::unique_ptr<Subexpr> clone() const override
+    {
+        return make_subexpr<Columns<Lst<String>>>(*this);
+    }
+
+    bool has_search_index() const final
+    {
+        auto target_table = m_link_map.get_target_table();
+        return target_table->search_index_type(m_column_key) == IndexType::General;
+    }
+
+    std::vector<ObjKey> find_all(Mixed value) const final
+    {
+        std::vector<ObjKey> ret;
+        std::vector<ObjKey> result;
+
+        StringIndex* index = m_link_map.get_target_table()->get_search_index(m_column_key);
+        REALM_ASSERT(index);
```

Comment:
I'm a bit wary of this. Is there really no path through the query engine that executes find_all without being indexed?

Reply:
It is only used in `Compare<TCond>::init`, and here it is guarded by a call to `has_search_index()`.